### PR TITLE
feat: add `save_on_fmt` setup option

### PR DESCRIPTION
**Pull request by ej-shafran**
Relevant to #8 

Couldn't figure out a simple way to add `save_on_fmt` as an option in `setup` without making a giant mess of things, settled on adding a `GuardFmtNoSave` command that just calls `do_fmt` with `false` for a `save_on_fmt` parameter.

### CODE REVIEW

**Comment by glepnir** [comment]
could we add a new option for this ?

**Comment by ej-shafran** [comment]
> could we add a new option for this ?

To `setup`? I could try...

**Comment by ej-shafran** [comment]
It's been a while, but I've added it as a configuration option in `setup`... I'll update it in the docs in a bit

File: lua/guard/init.lua [hunk 1]
```diff
@@ -2,6 +2,10 @@ local util = require('guard.util')
 local ft_handler = require('guard.filetype')
 local events = require('guard.events')
 
+local config = {
```
Comment (xiaoshihou514) [inline lua/guard/init.lua:5]: Hmm, I wonder why did you nest it in another table?

Comment (ej-shafran) [reply lua/guard/init.lua:5]: Well, it caused an issue if I didn't have it nested - it wouldn't properly update it. I think the fact that it's returned, so the module's value has it set to nil or an empty table, and then once `setup` runs it didn't update it.

I was thinking about using the 

```lua
local M = {}

M.config = {}

function M.setup(opts)
  -- ...
end

return M
```

pattern, but I saw that the rest of the files in this plugin don't really use it...

Comment (xiaoshihou514) [reply lua/guard/init.lua:5]: OK, it's an awkward workaround, but I guess it's fine for now.

**Comment by xiaoshihou514** [comment]
Would you mind documenting the change?

**Comment by xiaoshihou514** [comment]
Cool, ideally we should probably add a test for this but I guess it's fine for now.

**Pull request merged**